drivers/gpio/gpio_mcux.c

```diff
@@ -32,19 +42,42 @@ struct gpio_mcux_data {
 	sys_slist_t callbacks;
 };
 
+#if defined(CONFIG_SOC_SERIES_IMXRT7XX) && defined(CONFIG_SOC_FAMILY_NXP_IMXRT)
+static uint32_t GPIO_GetInstance(GPIO_Type *base)
+{
+
+	uint32_t instance;
+	/* Find the instance index from base address mappings. */
+	for (instance = 0; instance < ARRAY_SIZE(s_gpioBases); instance++) {
+		if (s_gpioBases[instance] == base) {
+			break;
+		}
+	}
+
+	assert(instance < ARRAY_SIZE(s_gpioBases));
+
+	return instance;
+}
+#endif
+
 static int gpio_mcux_configure(const struct device *dev,
 			       gpio_pin_t pin, gpio_flags_t flags)
 {
 	const struct gpio_mcux_config *config = dev->config;
 	GPIO_Type *gpio_base = config->gpio_base;
+
+#if defined(CONFIG_SOC_SERIES_IMXRT7XX) && defined(CONFIG_SOC_FAMILY_NXP_IMXRT)
+	uint32_t port_no = GPIO_GetInstance(gpio_base);
```

Comment:
why do we need to calculate this at runtime? Can't we store the port number in the device config structure?